cmd/unifi_exporter/main.go

```diff
@@ -9,49 +9,69 @@ import (
 	"net/http"
 	"strings"
 	"time"
-
+	"strconv"
+	"io/ioutil"
+	"gopkg.in/yaml.v2"
 	"github.com/mdlayher/unifi"
 	"github.com/mdlayher/unifi_exporter"
 	"github.com/prometheus/client_golang/prometheus"
 )
 
+type Config struct {
+	Listen map[string]string `yaml:"listen"`
+	Unifi map[string]string `yaml:"unifi"`
+}
+
 const (
 	// userAgent is ther user agent reported to the UniFi Controller API.
 	userAgent = "github.com/mdlayher/unifi_exporter"
 )
 
-var (
-	telemetryAddr = flag.String("telemetry.addr", ":9130", "host:port for UniFi exporter")
-	metricsPath   = flag.String("telemetry.path", "/metrics", "URL path for surfacing collected metrics")
-
-	unifiAddr = flag.String("unifi.addr", "", "address of UniFi Controller API")
-	username  = flag.String("unifi.username", "", "username for authentication against UniFi Controller API")
-	password  = flag.String("unifi.password", "", "password for authentication against UniFi Controller API")
-
-	site     = flag.String("unifi.site", "", "[optional] description of site to collect metrics for using UniFi Controller API; if none specified, all sites will be scraped")
-	insecure = flag.Bool("unifi.insecure", false, "[optional] do not verify TLS certificate for UniFi Controller API (warning: please use carefully)")
-	timeout  = flag.Duration("unifi.timeout", 5*time.Second, "[optional] timeout for UniFi Controller API requests")
-)
-
 func main() {
+	var configFile = flag.String("config.file", "", "Relative path to config file yaml")
 	flag.Parse()
 
-	if *unifiAddr == "" {
-		log.Fatal("address of UniFi Controller API must be specified with '-unifi.addr' flag")
+	var config Config
+	source, err := ioutil.ReadFile(*configFile)
+	if err != nil {
+		panic(err)
+	}
+	err = yaml.Unmarshal(source, &config)
+	if err != nil {
+		panic(err)
+	}
+
+	listenAddr    := config.Listen["address"] + ":" + config.Listen["port"]
+	metricsPath   := config.Listen["metricspath"]
+	unifiAddr     := config.Unifi["address"] + ":" + config.Unifi["port"]
+	username      := config.Unifi["username"]
+	password      := config.Unifi["password"]
+	site          := config.Unifi["site"]
+	b, err        := strconv.ParseBool(config.Unifi["insecure"])
+	insecure      := b
+	t, err        := strconv.Atoi(config.Unifi["timeoutsecs"])
```

Comment:
This should remain as a duration. Use: https://golang.org/pkg/time/#ParseDuration.  No "timeoutSecs", just "timeout".

Reply:
Yes this was my mistake, will be resolved in the latest commit.